src/main/java/com/vmware/vipclient/i18n/VIPCfgFactory.java

```diff
@@ -4,106 +4,49 @@
-        } else if (!prop.containsKey("productName")) {
-            throw new VIPClientInitException("Can't initialize VIPCfg. Product name is not defined.");
-        }
-        String productName = prop.getString("productName");
-        VIPCfgWrapper cfgWrapper = getCfg(productName, isGlobalCfg);
-        cfgWrapper.getVipCfg().initialize(cfgFile);
-        return cfgWrapper;
-    }
-
-    public static VIPCfgWrapper initialize(String cfgFile) throws VIPClientInitException {
-        return initialize(cfgFile, false);
-    }
-
-    public static VIPCfgWrapper getCfg(String productName, boolean isGlobalCfg) {
-        if (!contains(productName)) {
-            synchronized (ConfigsHolder.configs) {
-                if (!contains(productName)) {
-                    VIPCfgWrapper cfgWrapper = isGlobalCfg ? GlobalCfgHolder.globalCfg : new VIPCfgWrapper(new VIPCfg());
-                    cfgWrapper.setProductName(productName);
-                    addCfg(cfgWrapper);
-                }
-            }
-        }
-        return ConfigsHolder.configs.get(productName);
-    }
-
-    public static VIPCfgWrapper getCfg(String productName) {
-        return getCfg(productName, false);
-    }
-
-    private static void addCfg(VIPCfgWrapper vipCfgWrapper) {
-        ConfigsHolder.configs.put(vipCfgWrapper.getProductName(), vipCfgWrapper);
-    }
-
-    private static boolean contains(String productName) {
-        return ConfigsHolder.configs.containsKey(productName);
-    }
-
-    public static class VIPCfgWrapper implements VIPConfig {
-        private String productName;
-        private VIPCfg vipCfg;
-
-        private VIPCfgWrapper(VIPCfg vipCfg) {
-            this.vipCfg = vipCfg;
-        }
-        public String getProductName() {
-            return productName;
-        }
-
-        private void setProductName(String productName) {
-            this.productName = productName;
-        }
-
-        public VIPCfg getVipCfg() {
-            return vipCfg;
-        }
-    }
+	private static VIPCfgFactory inst = createInstance();
+	private static final String MAIN_PRODUCT_NAME = "mainProductName";
+
+	private final VIPCfg mainCfg = new VIPCfg();
+	private final Map<String, VIPCfg> configs = new ConcurrentHashMap<>();
+
+	private static VIPCfgFactory createInstance() {
+		VIPCfgFactory newInst = new VIPCfgFactory();
+		newInst.mainCfg.setProductName(MAIN_PRODUCT_NAME);
+		newInst.configs.put(newInst.mainCfg.getProductName(), newInst.mainCfg);
+		return newInst;
+	}
+
+	public static VIPCfg getMainCfg() {
+		return inst.mainCfg;
+	}
+
+	public static VIPCfg getCfg(String productName) {
+		synchronized (inst.configs) {
+			VIPCfg cfg = inst.configs.get(productName);
+			if (cfg != null) {
+				return cfg;
+			}
+
+			VIPCfg newCfg = new VIPCfg();
+			newCfg.setProductName(productName);
+			inst.configs.put(productName, newCfg);
+			return newCfg;
+		}
+	}
+
+	public static void changeProductName(VIPCfg cfg, String oldName) {
```

Comment:
Why do we want to provide the capability to change a product name? The application running at runtimr should not be changing product names.

Reply:
Yes. I agree with you. The product name can't be changed during lifetime. Here is to make sure change of product name won't be a problem because it's still public now.

Marked this as Deprecated.

Reply:
Let's make this protected then.